.github/workflows/ci-build.yml

```diff
@@ -1,31 +1,31 @@
-# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
-# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
-
-name: Node.js CI
+name: Node.js Build and Test
 
 on:
   push:
     branches:
       - main
   pull_request:
-  
+
 jobs:
   build:
-
     runs-on: ubuntu-latest
     timeout-minutes: 10
-
     strategy:
       matrix:
-        node-version: [14.x, 16.x, 18.x, 20.x]
-
+        node-version: [18.x, 20.x]
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
+      uses: actions/setup-node@v3
       with:
         node-version: ${{ matrix.node-version }}
-    - run: npm install
-    - run: npm run setup
-    - run: npm run test
-    # - run: npm run coverage
+    - name: Get Development Dependencies
+      run: npm i
+    - name: Build and Run Tests in Each Package
+      run: |
+        for pkg in packages/logger packages/oauth packages/rtm-api packages/socket-mode packages/types packages/web-api packages/webhook; do
+          pushd ${pkg}
+          npm install || npm list
```

Comment:
Running `npm install` automatically runs the typescript builds ( 🤢 ) so in case this fails, which can happen very easily if your typescript dependencies drift or are different than on your own machine, this line will run `npm list` to output all installed dependencies and their versions. This helped me identify the dependency difference I was experiencing and why I had no trouble running builds on my local machine but GitHub CI did have errors doing so.